core/src/main/java/org/apache/accumulo/core/data/ResourceGroupId.java

```diff
@@ -43,16 +43,23 @@ public class ResourceGroupId extends AbstractId<ResourceGroupId> {
 
   private ResourceGroupId(String canonical) {
     super(canonical);
-    if (!GROUP_NAME_PATTERN.matcher(canonical).matches()) {
-      throw new IllegalArgumentException(
-          "Group name: " + canonical + " contains invalid characters");
+    validateGroupName(canonical);
+  }
+
+  /**
+   * @throws IllegalArgumentException if the group name is invalid
+   */
+  public static void validateGroupName(String groupName) {
+    if (!GROUP_NAME_PATTERN.matcher(groupName).matches()) {
+      throw new IllegalArgumentException("Group name '" + groupName
+          + "' is invalid. Valid names must match the pattern: " + GROUP_NAME_PATTERN.pattern());
```

Comment:
Probably best to not add this to the public API. Instead, in the ClusterConfigParser, could just try to construct using `ResourceGroupId.of(id)`.

Reply:
In the current changes we swapped GROUP_NAME_PATTERN from public to private and instead created a new public method that uses GROUP_NAME_PATTERN internally. I feel like its its a slight improvement to have a public method to validate compared to a public `Pattern` object. Maybe its best to have neither public tough which I think is your point here, @ctubbsii

Reply:
fixed in 581fec7

Reply:
Having a public method that only does validation would be useful if there was a performance need to avoid allocating a object.  However w/ the caching that is done, that probably mitigates any problems w/ calling `of()` frequently as that avoids allocation.